lldb/include/lldb/Core/SwiftScratchContextReader.h

```diff
@@ -97,27 +47,25 @@ struct ScopedSharedMutexReader {
 /// or even separate scratch contexts for each lldb::Module. But it
 /// will only do this if no client holds on to a read lock on \b
 /// m_scratch_typesystem_lock.
-class SwiftScratchContextReader : ScopedSharedMutexReader {
-  TypeSystemSwiftTypeRefForExpressions *m_ptr;
+class SwiftScratchContextReader {
+  std::shared_lock<std::shared_mutex> m_lock;
+  TypeSystemSwiftTypeRefForExpressions *m_ts;
 
 public:
-  SwiftScratchContextReader(SharedMutex &mutex,
-                            TypeSystemSwiftTypeRefForExpressions &ctx)
-      : ScopedSharedMutexReader(&mutex), m_ptr(&ctx) {
-    assert(m_ptr && "invalid context");
-  }
-
-  TypeSystemSwiftTypeRefForExpressions *get() {
-    assert(m_ptr && "invalid context");
-    return m_ptr;
-  }
-
+  SwiftScratchContextReader(std::shared_mutex &mutex,
+                            TypeSystemSwiftTypeRefForExpressions &ts);
+  SwiftScratchContextReader(const SwiftScratchContextReader &) = delete;
+  SwiftScratchContextReader(SwiftScratchContextReader &&other) = default;
+  SwiftScratchContextReader &
+  operator=(SwiftScratchContextReader &&other) = default;
+  TypeSystemSwiftTypeRefForExpressions *get() { return m_ts; }
   TypeSystemSwiftTypeRefForExpressions *operator->() { return get(); }
   TypeSystemSwiftTypeRefForExpressions &operator*() { return *get(); }
 };
 
 /// An RAII object that just acquires the reader lock.
-struct SwiftScratchContextLock : ScopedSharedMutexReader {
+struct SwiftScratchContextLock {
+  std::shared_lock<std::shared_mutex> lock;
```

Comment:
maybe `m_lock` for consistency?

Reply:
IIUC the lldb coding style uses `m_` only in classes, not in structs.

Reply:
TIL